scripts/fetchUsers.js

```diff
@@ -0,0 +1,123 @@
+const fetchUsers = async () => {
+  try {
+    // fetch all pairs
+    const pairsList = await fetchAllPairs();
+
+    // Aggregate all pairs info
+    const pairsInfo = pairsList.map((pair) => ({
+      pairAddress: pair.address,
+      lastOrderId: pair.lastOrderId,
+    }));
+
+    // fetch all orders for all pairs
+    const allOrdersByPair = await Promise.all(
+      pairsInfo.map((pair) => {
+        const { pairAddress, lastOrderId } = pair;
+        const orderIds = Array.from({ length: lastOrderId }, (_, i) => i + 1); // [1, ..., lastOrderId]
+
+        return fetchOrdersByPair(pairAddress, orderIds);
+      })
+    );
+
+    // Process orders to count wallet addresses
+    const usersFreqCounter = allOrdersByPair.reduce((acc, curr) => {
+      if (!curr?.orders?.length) return acc;
+
+      // process all orders for each pair
+      const orders = curr.orders;
+
+      for (const order of orders) {
+        const radixWalletAddress = order?.settlementAccount; // account address for this order
+
+        if (!radixWalletAddress) continue;
+        else {
+          acc[radixWalletAddress] = (acc[radixWalletAddress] || 0) + 1;
+        }
+      }
+      return acc;
+    }, {});
+
+    // get total unique users count
+    const totalUsersCount = Object.values(usersFreqCounter).reduce(
+      (acc, curr) => {
+        return acc + curr;
```

Comment:
Will this not count the total number of orders, rather than the total number of unique users (accounts)? Or maybe I am understanding it wrong.
To get the number of accounts/users you can just take Array.from(Object.values(usersFreqCounter)).length

Reply:
Array.from(Object.values(usersFreqCounter)).length
Will this sum up all values for all the keys ? I believe it sums up the number of key/value pairs ? as would do Array.from(Object.keys(usersFreqCounter)).length

usersFreqCounter is an object with keys being unique accounts, and values being the sum of all orders passed by each corresponding unique account
so my totalUsersCount just sum up, for all unique accounts, all orders

Reply:
I agree with @fliebenberg , here's a recap:
```js
// To count all users
const totalUsersCount = Object.keys(usersFreqCounter).length;

// To sum all counts of orders placed
const totalOrdersCount = Object.values(usersFreqCounter).reduce((a,b) => a+b);
```

Reply:
Yes. my big mistake.. I will fix this, and we actually don t need the frequency counter..